recipes/reflect-cpp/all/conanfile.py

```diff
@@ -46,50 +67,76 @@ def _compilers_minimum_version(self):
             "apple-clang": "15",
         }
 
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
     def layout(self):
-        basic_layout(self, src_folder="src")
+        cmake_layout(self)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
 
     def requirements(self):
+        if self.options.with_cbor:
+            self.requires("tinycbor/0.6.0", transitive_headers=True)
+        if self.options.with_flatbuffers:
+            self.requires("flatbuffers/23.5.26", transitive_headers=True)
         if self.options.with_json:
             self.requires("yyjson/0.8.0", transitive_headers=True)
+        if self.options.with_msgpack:
+            self.requires("msgpack-c/6.0.0", transitive_headers=True)
+        if self.options.with_toml:
+            self.requires("tomlplusplus/3.4.0", transitive_headers=True)
         if self.options.with_xml:
             self.requires("pugixml/1.14", transitive_headers=True)
-        if self.options.with_flatbuffers:
-            self.requires("flatbuffers/23.5.26", transitive_headers=True)
         if self.options.with_yaml:
             self.requires("yaml-cpp/0.8.0", transitive_headers=True)
-        if self.options.with_msgpack:
-            self.requires("msgpack-c/6.0.0", transitive_headers=True)
-
-        if Version(self.version) >= "0.11.1":
-            self.requires("ctre/3.9.0", transitive_headers=True)
-
-    def package_id(self):
-        self.info.clear()
-
-    def validate(self):
-        if self.settings.get_safe("compiler.cppstd"):
-            check_min_cppstd(self, self._min_cppstd)
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(
-                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-            )
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
-        copy(self, pattern="LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(
+            self,
+            pattern="LICENSE*",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        copy(
```

Comment:
What is this used for? Why is the CMakeLists needed when consuming this package?

Reply:
Well, how are supposed to build the package without the CMakeLists.txt?